commit 3630167e1e49b2be64e244b4a8af1e81af014b84
Author: Boyuan Yang <073plan@gmail.com>
Date:   Wed Feb 7 01:06:07 2018 +0800

    src: Engine.cc: Fix a critical typo that leads to crashing
    
    $ cppcheck -j1 --quiet -f .
    [src/Engine.cc:512]: (error) Array 'KingMoveTab[4]' accessed at index 7, which is out of bounds.

diff --git a/src/Engine.cc b/src/Engine.cc
index 0ebc18d..d8782e0 100644
--- a/src/Engine.cc
+++ b/src/Engine.cc
@@ -517,7 +517,7 @@ bool Engine::mate()
 				case RED_KNIGHT:
 				case BLACK_KNIGHT:
 					for(int k=0;k<8;k++){
-						dst=src+ KingMoveTab[k];
+						dst=src+ KnightMoveTab[k];
 						if(in_board(dst)){
 							int mv = get_move(src,dst);
 							if(make_move(mv)){
